service/replay.go

```diff
@@ -92,11 +93,25 @@ func (c *ReplayCache) Add(id string, salt []byte) bool {
 		return false
 	}
 	_, inArchive := c.archive[hash]
-	if len(c.active) == c.capacity {
+	if len(c.active) >= c.capacity {
```

Comment:
@fortuna I already merged a bit too fast, but just wanted to point out this specific change I made as part of not actively shrinking the maps in the `Resize()` method. This is needed, otherwise we'll never shrink maps that were resized down.